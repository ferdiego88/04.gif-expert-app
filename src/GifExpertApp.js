import { useState } from "react";
import  AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

export const GifExpertApp = ({defaultCategory =[] }) => {

    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategory);

    // const handleAdd = () => {
    //     setCategories(
    //         [...categories,'HunterXHunter']
    //     );

    //     // O de esta forma
    //     //setCategories(category => [...category,'HunterXHunter']);
    // }

    return (
        <>
            <h2>
                Beffis is Back App (Flavia, Geyvin, JheXson,PingoSanti)
            </h2>
            <AddCategory setCategories={setCategories}/>
            <hr></hr>

            <ol>
                {
                    categories.map( category => 
                    <GifGrid
                    key={category}
                    category={category} 
                    />
                    )
                }

            </ol>
        </>
    );
    
}

export default GifExpertApp;