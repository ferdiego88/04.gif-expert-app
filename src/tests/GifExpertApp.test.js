import { shallow }from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('Test in GifExpert App',() => {

    let wrapper = shallow(<GifExpertApp />)

    test('It should show up correctly.', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('a list of categories should be displayed', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaultCategory={categories}/>);

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})