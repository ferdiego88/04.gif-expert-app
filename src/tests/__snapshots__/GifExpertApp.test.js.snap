// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpert App It should show up correctly. 1`] = `
<Fragment>
  <h2>
    Beffis is Back App (Flavia, Geyvin, JheXson,PingoSanti)
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in GifExpert App a list of categories should be displayed 1`] = `
<Fragment>
  <h2>
    Beffis is Back App (Flavia, Geyvin, JheXson,PingoSanti)
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
